### PR TITLE
refactor(autoware_universe_utils): changed the API to be more intuitive and added documentation

### PR DESCRIPTION
**Pull request by yhisaki**
## Description

In the current implementation, `takeData()` is used to receive the latest data, `takeNewData()` is used to receive new data, and `InterProcessPollingSubscriber<T, N>` is used to receive historical data. However, this implementation has the following issues:
1. It is not good for a single class to have both the `takeData()` and `takeNewData()` member functions. Essentially, objects that use `takeData()` will never call `takeNewData()`, and objects that use `takeNewData()` will never call `takeData()`.
2. The value of N in `InterProcessPollingSubscriber<T, N>` is meaningless for N >= 2 in terms of implementation.

In this PR, the implementation has been changed to be more intuitive by passing the following PollingPolicy as a template:

```cpp
InterProcessPollingSubscriber<MessageT, polling_policy::Latest> sub_topic_{this, "~/topic_name"};
auto data = sub_topic_.takeData(); // Same as the previous implementation.

InterProcessPollingSubscriber<MessageT, polling_policy::Newest> sub_topic_{this, "~/topic_name"};
auto data = sub_topic_.takeData(); // Same as the `takeNewData()` in the previous implementation.

InterProcessPollingSubscriber<MessageT, polling_policy::All> sub_topic_{this, "~/topic_name"};
auto data = sub_topic_.takeData(); // It returns the data in the buffer as an `std::vector`.
```

## Tests performed

Tested on psim.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: evaluator/autoware_control_evaluator/include/autoware/control_evaluator/control_evaluator_node.hpp [hunk 1]
```diff
@@ -79,14 +79,16 @@ class ControlEvaluatorNode : public rclcpp::Node
...
+    LaneletRoute, autoware::universe_utils::polling_policy::Newest>
+    route_subscriber_{this, "~/input/route", rclcpp::QoS{1}.transient_local()};
+  autoware::universe_utils::InterProcessPollingSubscriber<
+    LaneletMapBin, autoware::universe_utils::polling_policy::Newest>
+    vector_map_subscriber_{this, "~/input/vector_map", rclcpp::QoS{1}.transient_local()};
 
   rclcpp::Publisher<DiagnosticArray>::SharedPtr metrics_pub_;
 
-  // update Route Handler
+  // update Route Handleru
```
Comment (maxime-clem) [inline evaluator/autoware_control_evaluator/include/autoware/control_evaluator/control_evaluator_node.hpp:91]:
```suggestion
  // update Route Handler
```

**Review by maxime-clem — APPROVED** [review]

File: evaluator/autoware_control_evaluator/include/autoware/control_evaluator/control_evaluator_node.hpp (hunk 1 above)
Comment (yhisaki) [reply evaluator/autoware_control_evaluator/include/autoware/control_evaluator/control_evaluator_node.hpp]: thanks!!

**Review by soblin — APPROVED** [review]

**Review by kyoichi-sugahara — APPROVED** [review]

**Review by satoshi-ota — APPROVED** [review]

File: evaluator/autoware_control_evaluator/include/autoware/control_evaluator/control_evaluator_node.hpp (hunk 1 above)
Comment (danielsanchezaran) [inline evaluator/autoware_control_evaluator/include/autoware/control_evaluator/control_evaluator_node.hpp:91]:
```suggestion
  // update Route Handler
```

**Review by danielsanchezaran — APPROVED** [review]
Please fix the  typo. The rest LGTM

**Pull request merged**